### ISSUE TITLE
cat /dev/urandom crashes alacritty (again)

### ISSUE THREAD

**Issue by Timidger** [jwilm/alacritty#499]
While testing alacritty on Way Cooler, I ran into this issue after running the simple stress test of `cat /dev/urandom`. This was on the latest master, downloaded from the AUR package.

Error output:

```
num_cols, num_lines = 135, 64
[unhandled] esc_dispatch params=[], ints=[], byte='4' (34)
[unhandled] esc_dispatch params=[], ints=[], byte='9' (39)
[unhandled] esc_dispatch params=[], ints=[], byte='j' (6a)
[unhandled] esc_dispatch params=[], ints=[], byte='m' (6d)
[unhandled] esc_dispatch params=[], ints=[], byte='<' (3c)
[unhandled] esc_dispatch params=[], ints=[], byte='N' (4e)
[unhandled] esc_dispatch params=[], ints=[], byte='f' (66)
[Unhandled CSI] action='}', args=[], intermediates=[]
[unhandled] esc_dispatch params=[], ints=[], byte='a' (61)
[unhandled] esc_dispatch params=[], ints=[], byte='I' (49)
[unhandled] esc_dispatch params=[], ints=[], byte='q' (71)
[unhandled] esc_dispatch params=[], ints=[], byte='1' (31)
[unhandled] esc_dispatch params=[], ints=[], byte='k' (6b)
[unhandled] esc_dispatch params=[], ints=[], byte='R' (52)
[unhandled] esc_dispatch params=[], ints=[], byte='x' (78)
[unhandled] esc_dispatch params=[], ints=[], byte='4' (34)
[unhandled] esc_dispatch params=[], ints=[], byte='I' (49)
[unhandled] esc_dispatch params=[], ints=[], byte='p' (70)
[unhandled] esc_dispatch params=[], ints=[], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[], byte='A' (41)
[unhandled] esc_dispatch params=[], ints=[], byte='F' (46)
[unhandled] esc_dispatch params=[], ints=[], byte='o' (6f)
[unhandled] esc_dispatch params=[], ints=[32], byte='J' (4a)
[unhandled] esc_dispatch params=[], ints=[], byte='I' (49)
thread 'main' panicked at 'index 61 out of range for slice of length 60', /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libcore/slice.rs:578
stack backtrace:
   1:     0x557f968bec8a - <unknown>
   2:     0x557f968be111 - <unknown>
   3:     0x557f968bd270 - <unknown>
   4:     0x557f968bcdc1 - <unknown>
   5:     0x557f968bcd39 - <unknown>
   6:     0x557f968d2606 - <unknown>
   7:     0x557f968d2423 - <unknown>
   8:     0x557f9689daf4 - <unknown>
   9:     0x557f9689d6a0 - <unknown>
  10:     0x557f967bc760 - <unknown>
  11:     0x557f96713366 - <unknown>
  12:     0x557f9670b4d6 - <unknown>
  13:     0x557f96719a80 - <unknown>
  14:     0x7fc8646c3510 - __libc_start_main
  15:     0x557f966d9b09 - <unknown>
  16:                0x0 - <unknown>
[unhandled] esc_dispatch params=[], ints=[], byte='R' (52)
[unhandled] esc_dispatch params=[], ints=[], byte='m' (6d)
[unhandled] esc_dispatch params=[], ints=[], byte='@' (40)
[unhandled] esc_dispatch params=[], ints=[], byte='k' (6b)
[unhandled] esc_dispatch params=[], ints=[], byte='r' (72)
[unhandled] esc_dispatch params=[], ints=[], byte='u' (75)
[unhandled] esc_dispatch params=[], ints=[], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[], byte='U' (55)
[unhandled] esc_dispatch params=[], ints=[], byte='I' (49)
[unhandled] esc_dispatch params=[], ints=[], byte='J' (4a)
[unhandled] esc_dispatch params=[], ints=[], byte='2' (32)
[unhandled] esc_dispatch params=[], ints=[], byte='~' (7e)
[unhandled] esc_dispatch params=[], ints=[], byte='R' (52)
[unhandled] esc_dispatch params=[], ints=[], byte='|' (7c)
[unhandled] esc_dispatch params=[], ints=[], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[], byte='o' (6f)
[unhandled] esc_dispatch params=[], ints=[], byte='v' (76)
[unhandled] esc_dispatch params=[], ints=[], byte='J' (4a)
[unhandled] esc_dispatch params=[], ints=[], byte='N' (4e)
[unhandled] esc_dispatch params=[], ints=[], byte='h' (68)
[unhandled] esc_dispatch params=[], ints=[], byte='p' (70)
[unhandled] esc_dispatch params=[], ints=[], byte='<' (3c)
[unhandled] esc_dispatch params=[], ints=[], byte='q' (71)
[unhandled] esc_dispatch params=[], ints=[], byte='|' (7c)
[unhandled] esc_dispatch params=[], ints=[], byte='F' (46)
[unhandled] esc_dispatch params=[], ints=[], byte='n' (6e)
[unhandled] esc_dispatch params=[], ints=[], byte='h' (68)
[unhandled] esc_dispatch params=[], ints=[35], byte='m' (6d)
[unhandled] esc_dispatch params=[], ints=[], byte='l' (6c)
[unhandled] esc_dispatch params=[], ints=[], byte='o' (6f)
[unhandled] esc_dispatch params=[], ints=[40], byte='_' (5f)
[unhandled] esc_dispatch params=[], ints=[40, 44], byte='V' (56)
[unhandled] esc_dispatch params=[], ints=[], byte='<' (3c)
[unhandled] esc_dispatch params=[], ints=[], byte='k' (6b)
[unhandled] esc_dispatch params=[], ints=[], byte='L' (4c)
[unhandled] esc_dispatch params=[], ints=[], byte='O' (4f)
[unhandled] esc_dispatch params=[], ints=[], byte='u' (75)
[unhandled] esc_dispatch params=[], ints=[], byte='1' (31)
[unhandled] esc_dispatch params=[], ints=[], byte='y' (79)
[unhandled] esc_dispatch params=[], ints=[], byte='i' (69)
[unhandled] esc_dispatch params=[], ints=[], byte='f' (66)
[unhandled] esc_dispatch params=[], ints=[], byte='C' (43)
[unhandled] esc_dispatch params=[], ints=[], byte='u' (75)
[unhandled] esc_dispatch params=[], ints=[41], byte='y' (79)
[unhandled] esc_dispatch params=[], ints=[], byte='Q' (51)
[unhandled] esc_dispatch params=[], ints=[], byte='d' (64)
[unhandled] esc_dispatch params=[], ints=[], byte='r' (72)
[unhandled] esc_dispatch params=[], ints=[45, 39], byte='k' (6b)
[unhandled] esc_dispatch params=[], ints=[], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[], byte='5' (35)
[unhandled] esc_dispatch params=[], ints=[], byte='J' (4a)
[unhandled] esc_dispatch params=[], ints=[], byte='W' (57)
[unhandled] esc_dispatch params=[], ints=[], byte='{' (7b)
[unhandled] esc_dispatch params=[], ints=[], byte='K' (4b)
[unhandled] esc_dispatch params=[], ints=[], byte='4' (34)
[unhandled] esc_dispatch params=[], ints=[], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[40], byte='b' (62)
[unhandled] esc_dispatch params=[], ints=[], byte='n' (6e)
[unhandled] esc_dispatch params=[], ints=[], byte='C' (43)
[unhandled] esc_dispatch params=[], ints=[41], byte='J' (4a)
[unhandled] esc_dispatch params=[], ints=[34, 41], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[37], byte='x' (78)
[unhandled] esc_dispatch params=[], ints=[], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[], byte='a' (61)
[unhandled] esc_dispatch params=[], ints=[], byte='t' (74)
[unhandled] esc_dispatch params=[], ints=[45], byte='L' (4c)
[unhandled] esc_dispatch params=[], ints=[44], byte='1' (31)
[unhandled] esc_dispatch params=[], ints=[], byte='p' (70)
[unhandled] esc_dispatch params=[], ints=[], byte='5' (35)
[unhandled] esc_dispatch params=[], ints=[], byte='9' (39)
[unhandled] esc_dispatch params=[], ints=[44], byte='g' (67)
[unhandled] esc_dispatch params=[], ints=[35], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[42], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[], byte='~' (7e)
[unhandled] esc_dispatch params=[], ints=[], byte='3' (33)
[unhandled] esc_dispatch params=[], ints=[], byte='<' (3c)
[unhandled] esc_dispatch params=[], ints=[], byte='<' (3c)
[unhandled] esc_dispatch params=[], ints=[33], byte='1' (31)
[unhandled] esc_dispatch params=[], ints=[], byte='I' (49)
[unhandled] esc_dispatch params=[], ints=[], byte='e' (65)
[unhandled] esc_dispatch params=[], ints=[45], byte='g' (67)
[unhandled] esc_dispatch params=[], ints=[], byte=':' (3a)
[unhandled] esc_dispatch params=[], ints=[], byte='b' (62)
[unhandled] esc_dispatch params=[], ints=[], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[], byte='6' (36)
[unhandled] esc_dispatch params=[], ints=[47], byte='z' (7a)
[unhandled] esc_dispatch params=[], ints=[], byte='G' (47)
[unhandled] esc_dispatch params=[], ints=[], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[40], byte='x' (78)
[unhandled] esc_dispatch params=[], ints=[], byte='o' (6f)
[unhandled] esc_dispatch params=[], ints=[], byte='l' (6c)
[unhandled] esc_dispatch params=[], ints=[36], byte='L' (4c)
[unhandled osc_dispatch]: [['ò','|','s',',','K','<','þ','`','1','+','4','!','¸','\u{87}','Ø','\u{81}','l',' ','U','5','Ó','<','Ô','×','d','÷','ÿ','ß','ý','\u{9e}','ú','º','à','¥','É','½','Ó','v','<','X','a','G','b','ÿ','ÿ','ü','¢','¸','C','Ò','Ë','9','µ','r','Û','Ó','\u{98}','\u{8e}','\u{8e}','§','¡','5','ê','\u{9c}','ê','½','e','\u{87}','\'','\\','#','ä','\u{8c}','\u{99}','}','ë','\u{9e}','µ','\u{88}','+','Â','p','ö','a','ç','\u{87}','&','´','ú','Å','þ','Ã','\\','g','\u{9b}','»','$','î','Î','\u{97}','\u{a0}','\u{92}','\u{94}','©','^','Ó','Õ','Ú','\"','\u{88}','\u{8b}','¨','\u{80}','Ó','¸','p','r','x','\u{83}','\u{99}','¸','\u{92}','T','ø','H','C','É','Ø','%','b','\u{80}','\u{89}','Ñ','$','4','{','a','ù','Ú','¾','ù',],] at line 644
[unhandled] esc_dispatch params=[], ints=[35], byte='N' (4e)
[unhandled] esc_dispatch params=[], ints=[47], byte='Y' (59)
[unhandled] esc_dispatch params=[], ints=[], byte='U' (55)
[unhandled] esc_dispatch params=[], ints=[], byte='r' (72)
[unhandled] esc_dispatch params=[], ints=[], byte='Y' (59)
[unhandled] esc_dispatch params=[], ints=[43], byte='5' (35)
[unhandled] esc_dispatch params=[], ints=[], byte='w' (77)
[unhandled] esc_dispatch params=[], ints=[], byte='O' (4f)
[unhandled] esc_dispatch params=[], ints=[37, 39], byte='^' (5e)
[unhandled] esc_dispatch params=[], ints=[42], byte='2' (32)
[unhandled] esc_dispatch params=[], ints=[], byte='n' (6e)
[unhandled] esc_dispatch params=[], ints=[], byte='m' (6d)
[unhandled] esc_dispatch params=[], ints=[], byte='{' (7b)
[unhandled] esc_dispatch params=[], ints=[], byte='L' (4c)
[unhandled] esc_dispatch params=[], ints=[], byte='2' (32)
[unhandled] esc_dispatch params=[], ints=[], byte='`' (60)
[unhandled] esc_dispatch params=[], ints=[], byte='<' (3c)
[unhandled] esc_dispatch params=[], ints=[], byte='o' (6f)
[unhandled] esc_dispatch params=[], ints=[], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[], byte='e' (65)
[unhandled] esc_dispatch params=[], ints=[], byte='5' (35)
[unhandled] esc_dispatch params=[], ints=[], byte='}' (7d)
[unhandled] esc_dispatch params=[], ints=[], byte='W' (57)
[unhandled] esc_dispatch params=[], ints=[], byte='6' (36)
[unhandled] esc_dispatch params=[], ints=[34], byte='0' (30)
[unhandled] esc_dispatch params=[], ints=[], byte='{' (7b)
[unhandled] esc_dispatch params=[], ints=[], byte='a' (61)
[unhandled] esc_dispatch params=[], ints=[], byte='C' (43)
[unhandled] esc_dispatch params=[], ints=[], byte='K' (4b)
[unhandled] esc_dispatch params=[], ints=[], byte='b' (62)
[unhandled] esc_dispatch params=[], ints=[], byte='W' (57)
[unhandled] esc_dispatch params=[], ints=[32], byte='K' (4b)
[unhandled] esc_dispatch params=[], ints=[], byte='1' (31)
[unhandled] esc_dispatch params=[], ints=[], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[], byte='o' (6f)
[unhandled] esc_dispatch params=[], ints=[], byte='2' (32)
[unhandled] esc_dispatch params=[], ints=[], byte='k' (6b)
[unhandled] esc_dispatch params=[], ints=[33], byte='_' (5f)
[unhandled] esc_dispatch params=[], ints=[], byte='2' (32)
[unhandled] esc_dispatch params=[], ints=[], byte='2' (32)
[unhandled] esc_dispatch params=[], ints=[], byte='5' (35)
[unhandled] esc_dispatch params=[], ints=[], byte='W' (57)
[unhandled] esc_dispatch params=[], ints=[], byte='p' (70)
[unhandled] esc_dispatch params=[], ints=[40], byte='F' (46)
[unhandled] esc_dispatch params=[], ints=[44], byte='1' (31)
[unhandled] esc_dispatch params=[], ints=[], byte='h' (68)
[unhandled] esc_dispatch params=[], ints=[43], byte='l' (6c)
[unhandled] esc_dispatch params=[], ints=[38, 36], byte='t' (74)
[unhandled] esc_dispatch params=[], ints=[], byte='a' (61)
[unhandled] esc_dispatch params=[], ints=[], byte='3' (33)
[unhandled] esc_dispatch params=[], ints=[], byte='{' (7b)
[unhandled] esc_dispatch params=[], ints=[], byte='3' (33)
[unhandled] esc_dispatch params=[], ints=[], byte='6' (36)
[unhandled] esc_dispatch params=[], ints=[], byte='T' (54)
[unhandled] esc_dispatch params=[], ints=[], byte='5' (35)
[unhandled] esc_dispatch params=[], ints=[], byte='1' (31)
[unhandled] esc_dispatch params=[], ints=[], byte='C' (43)
[unhandled] esc_dispatch params=[], ints=[45], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[47], byte='K' (4b)
[unhandled] esc_dispatch params=[], ints=[46], byte='~' (7e)
[unhandled] esc_dispatch params=[], ints=[42, 46], byte=';' (3b)
[unhandled] esc_dispatch params=[], ints=[], byte='I' (49)
[unhandled] esc_dispatch params=[], ints=[], byte='f' (66)
[unhandled] esc_dispatch params=[], ints=[], byte='q' (71)
[unhandled] esc_dispatch params=[], ints=[], byte='u' (75)
[unhandled] esc_dispatch params=[], ints=[], byte='3' (33)
[unhandled] esc_dispatch params=[], ints=[36], byte='J' (4a)
[unhandled] esc_dispatch params=[], ints=[40], byte='4' (34)
[unhandled] esc_dispatch params=[], ints=[], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[], byte='W' (57)
[unhandled] esc_dispatch params=[], ints=[44], byte='1' (31)
[unhandled] esc_dispatch params=[], ints=[], byte='i' (69)
[unhandled] esc_dispatch params=[], ints=[], byte='q' (71)
[unhandled] esc_dispatch params=[], ints=[], byte='|' (7c)
[unhandled] esc_dispatch params=[], ints=[], byte='`' (60)
[unhandled] esc_dispatch params=[], ints=[40], byte='I' (49)
[unhandled] esc_dispatch params=[], ints=[], byte='3' (33)
[unhandled] esc_dispatch params=[], ints=[], byte='L' (4c)
[unhandled] esc_dispatch params=[], ints=[], byte='i' (69)
[unhandled] esc_dispatch params=[], ints=[], byte='a' (61)
[unhandled] esc_dispatch params=[], ints=[], byte='{' (7b)
[unhandled] esc_dispatch params=[], ints=[], byte='V' (56)
[unhandled] esc_dispatch params=[], ints=[46], byte='0' (30)
[unhandled] esc_dispatch params=[], ints=[42, 34], byte='Y' (59)
[unhandled] esc_dispatch params=[], ints=[], byte='U' (55)
[unhandled] esc_dispatch params=[], ints=[], byte='w' (77)
[unhandled] esc_dispatch params=[], ints=[46], byte='@' (40)
[unhandled] esc_dispatch params=[], ints=[], byte='Q' (51)
[unhandled] esc_dispatch params=[], ints=[], byte='<' (3c)
[unhandled] esc_dispatch params=[], ints=[], byte='x' (78)
[unhandled] esc_dispatch params=[], ints=[45], byte='g' (67)
[unhandled] esc_dispatch params=[], ints=[], byte='N' (4e)
[unhandled] esc_dispatch params=[], ints=[42], byte='k' (6b)
[unhandled] esc_dispatch params=[], ints=[], byte='h' (68)
[unhandled] esc_dispatch params=[], ints=[], byte='C' (43)
[unhandled] esc_dispatch params=[], ints=[], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[], byte='@' (40)
```

**Comment by mdevlamynck** [comment]
Same issue here on archlinux, commit 8d4c10dad65a2a26c5801448ed82ed8ebef6850a.
Here is the log with RUST_BACKTRACE=1 and with a debug build to have the backtrace:
```
    Finished dev [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `target/debug/alacritty`
num_cols, num_lines = 128, 48
num_cols, num_lines = 80, 24
[unhandled] esc_dispatch params=[], ints=[35], byte='n' (6e)
[unhandled] esc_dispatch params=[], ints=[44], byte='b' (62)
[unhandled] esc_dispatch params=[], ints=[], byte='R' (52)
[unhandled] esc_dispatch params=[], ints=[], byte='a' (61)
[unhandled] esc_dispatch params=[], ints=[], byte='9' (39)
[unhandled] esc_dispatch params=[], ints=[], byte='w' (77)
[unhandled] esc_dispatch params=[], ints=[], byte='V' (56)
[unhandled] esc_dispatch params=[], ints=[], byte='<' (3c)
[unhandled] esc_dispatch params=[], ints=[], byte='h' (68)
[unhandled] esc_dispatch params=[], ints=[], byte='1' (31)
[unhandled osc_dispatch]: [['c','é','=','1','ç','c','q','À','6','ø','¨','E','°','<','¦',],] at line 644
[unhandled] esc_dispatch params=[], ints=[46], byte='r' (72)
[unhandled] esc_dispatch params=[], ints=[], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[], byte='K' (4b)
[unhandled] esc_dispatch params=[], ints=[], byte='A' (41)
[unhandled osc_dispatch]: [['è','Ï','ý','\u{8c}','\u{90}','7','\u{80}','\u{92}','$','$','\u{82}','«','b','Ö','^','\u{98}','d','J','r',',','d','×','Ð','¯','°','d','\u{81}','Å','²','C','r',']','a',],] at line 644
[unhandled] esc_dispatch params=[], ints=[33], byte='t' (74)
[unhandled] esc_dispatch params=[], ints=[], byte=':' (3a)
[unhandled] esc_dispatch params=[], ints=[], byte='4' (34)
[unhandled] esc_dispatch params=[], ints=[], byte='q' (71)
[unhandled] esc_dispatch params=[], ints=[], byte='0' (30)
[unhandled] esc_dispatch params=[], ints=[], byte='G' (47)
[unhandled] esc_dispatch params=[], ints=[], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[43], byte='|' (7c)
[unhandled] esc_dispatch params=[], ints=[], byte='4' (34)
[unhandled] esc_dispatch params=[], ints=[34], byte='K' (4b)
[unhandled] esc_dispatch params=[], ints=[], byte='y' (79)
thread 'main' panicked at 'index 56 out of range for slice of length 52', /checkout/src/libcore/slice/mod.rs:658
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: core::slice::slice_index_len_fail
             at /checkout/src/libcore/slice/mod.rs:658
   1: <core::ops::Range<usize> as core::slice::SliceIndex<T>>::index
             at /checkout/src/libcore/slice/mod.rs:789
   2: core::slice::<impl core::ops::Index<I> for [T]>::index
             at /checkout/src/libcore/slice/mod.rs:640
   3: font::ft::FreeTypeRasterizer::get_rendered_glyph
             at ./font/src/ft/mod.rs:217
   4: font::ft::FreeTypeRasterizer::get_rendered_glyph
             at ./font/src/ft/mod.rs:194
   5: <font::ft::FreeTypeRasterizer as font::Rasterize>::get_glyph
             at ./font/src/ft/mod.rs:85
   6: alacritty::renderer::GlyphCache::get::{{closure}}
             at ./src/renderer/mod.rs:270
   7: <std::collections::hash::map::Entry<'a, K, V>>::or_insert_with
             at /checkout/src/libstd/collections/hash/map.rs:1940
   8: alacritty::renderer::GlyphCache::get
             at ./src/renderer/mod.rs:267
   9: alacritty::renderer::RenderApi::render_cells
             at ./src/renderer/mod.rs:785
  10: alacritty::display::Display::draw::{{closure}}
             at ./src/display.rs:299
  11: alacritty::renderer::QuadRenderer::with_api
             at ./src/renderer/mod.rs:610
  12: alacritty::display::Display::draw
             at ./src/display.rs:297
  13: alacritty::run
             at ./src/main.rs:172
  14: alacritty::main
             at ./src/main.rs:64
[unhandled] esc_dispatch params=[], ints=[], byte='q' (71)
[unhandled] esc_dispatch params=[], ints=[], byte='t' (74)
[unhandled] esc_dispatch params=[], ints=[], byte='9' (39)
[unhandled] esc_dispatch params=[], ints=[], byte='`' (60)
[unhandled] esc_dispatch params=[], ints=[], byte='t' (74)
[unhandled] esc_dispatch params=[], ints=[], byte='2' (32)
[unhandled] esc_dispatch params=[], ints=[38], byte='Y' (59)
[unhandled] esc_dispatch params=[], ints=[], byte='|' (7c)
[unhandled] esc_dispatch params=[], ints=[33], byte='?' (3f)
[unhandled] esc_dispatch params=[], ints=[39], byte='2' (32)
[unhandled] esc_dispatch params=[], ints=[], byte='O' (4f)
[unhandled] esc_dispatch params=[], ints=[], byte='C' (43)
[unhandled] esc_dispatch params=[], ints=[36], byte='X' (58)
[unhandled] esc_dispatch params=[], ints=[], byte='A' (41)
[unhandled] esc_dispatch params=[], ints=[], byte='Y' (59)
[unhandled] esc_dispatch params=[], ints=[], byte='J' (4a)
[unhandled] esc_dispatch params=[], ints=[], byte='l' (6c)
[unhandled] esc_dispatch params=[], ints=[], byte='r' (72)
[unhandled] esc_dispatch params=[], ints=[], byte='q' (71)
[unhandled] esc_dispatch params=[], ints=[], byte='1' (31)
```

**Comment by mdevlamynck** [comment]
In ./font/src/ft/mod.rs I added the following assertion which fails:
```
        let bitmap = glyph.bitmap();
        let buf = bitmap.buffer();
        let pitch = bitmap.pitch().abs() as usize;

        let mut packed = Vec::with_capacity((bitmap.rows() * bitmap.width()) as usize);
        for i in 0..bitmap.rows() {
            let start = (i as usize) * pitch;
            let stop = start + bitmap.width() as usize;
            assert!(pitch > bitmap.width() as usize);        // <- added this
            packed.extend_from_slice(&buf[start..stop]);
        }
```
I don't know how freetype's bitmap works but this seems weird. [Bitmap's documentation](https://www.freetype.org/freetype2/docs/reference/ft2-basic_types.html#FT_Bitmap) states that the pitch is the size of a row and the width is the number of pixels in a row, so the width should be smaller if it's using the same unit (does a pixel equal to a u8?). If not, I don't undertsand how using width to index buf makes sense. I hope this helps.

**Comment by jwilm** [comment]
Fixed in #590
